plugin/cache/item.go

```diff
@@ -28,7 +28,7 @@ func newItem(m *dns.Msg, now time.Time, d time.Duration) *item {
 	i.Authoritative = m.Authoritative
 	i.AuthenticatedData = m.AuthenticatedData
 	i.RecursionAvailable = m.RecursionAvailable
-	i.Answer = m.Answer
+	i.Answer = append([]dns.RR{}, m.Answer...)
```

Comment:
Isn't a `copy` more efficient?

Same is true for the i.Ns part of the answer

Reply:
Good point, I have:

- Use make + copy, and deep copy the underlying Record
- Applied the same fix for `i.Ns`